### PR TITLE
Copy-edit the Effective Dart intro, clarify definitions

### LINKED ISSUE #77: Effective Dart, glossary feedback.
_From @lrhn on October 29, 2015 7:59_

Generally great stuff, but I fear the glossary may be a little too vague to be really useful.
Warning: This may be overly pedantic, but it looks like a _definition_, and I have very high standards for those :)

> A member is any definition or declaration in a library. This includes top-level variables, getters, setters, 
> and functions. It also includes instance and static members inside classes: fields, getters, setters,
> methods, and operators.

Since all declarations in Dart are inside a library, the first sentence means that this is the same as "declaration". I'm not sure what a "definition" is.
It is said to "include" top-level variables, getters, setters and functions. "Include" is non-exclusive, so that's basically just listing examples. It's exactly omitting type declarations (class, typedef, enum) which is conspicuous, especially since the class-contained members do the same.

So: is a class declaration a member?  (It is a declaration in a library, so by the above, it would be, but I think it isn't intended).

(It's not clear whether the _getter_ is a member (semantics) or the _getter declaration_ is a member (syntax), but I'll assume it's the getter, and the getter declaration is a "member declaration".).

How about:

 A _library member_ is a top-level field, getter, setter or function.
 A _class member_ is a (static or instance) field, getter, setter, function, or operator.
 A "member" may be either a library or class member.
 Types (declared by `class`, `typedef` or `enum`) are not considered "members" of the library, 
 neither are constructors members of the class.

(if that is what is intended?)

> A variable is any variable declaration, local or top-level. This also includes member parameters.

You haven't defined "variable" or "variable declaration". Traditionally I would not consider a "field" as a "variable", but I know that others do. Some may not accept a "final" as being "variable". 
This also conflates "variable" and "variable declaration" which I would distinguish as, respectively, semantics and syntax of the same thing (and then the glossary entry is really just saying that a variable is a variable).

How about:
"A variable is any top-level, static or instance field, any function parameter or local variable variable declared inside function bodies."
This excludes getter/setter pairs from being variables, which I hope is the intention - a variable is something that has a slot allocated that can hold a value.

> A type is any named type declaration: a class, typedef, or enum.

Again conflating type and type declaration. 
I'd prefer "A type is declared by a type declaration: a class, typedef or enum."
This omits that "dynamic" is also a type? (But "void" is not, it's only a type-expression (syntax) that does not correspond to a type (semantics)).

> A property is a “field-like” named construct. This includes actual fields inside classes, as well as 
> getters and setters. This also includes top level variables, getters, and setters

I would not use "property" for top-level things. A property is a property _of_ something.
I'd prefer "A property is an aspect of an object that has a name, like the `length` of a `List` or the `keys` of a `Map`".

_Copied from original issue: dart-lang/www.dartlang.org#1506_

### PR DESCRIPTION
**Pull request by munificent**
Finally trying to brush some of the dust off.

Fix #77.

### CODE REVIEW

**Comment by kwalrath** [comment]
@Sfshaza is out, so I can take a look unless you want to wait for her.

**Comment by munificent** [comment]
That would be great, thanks!

File: src/_guides/language/effective-dart/index.md [hunk 1]
```diff
@@ -14,24 +14,21 @@ nextpage:
 
 Over the past several years, we've written a ton of Dart code and learned a lot
 about what works well and what doesn't. We're sharing this with you so you can
-write consistent, robust, fast code too.
+write consistent, robust, fast code too. There are two over-arching themes:
```
Comment (kwalrath) [inline src/_guides/language/effective-dart/index.md]: over-arching -> overarching

File: src/_guides/language/effective-dart/index.md [hunk 2]
```diff
@@ -14,24 +14,21 @@ nextpage:
...
 
-There are two over-arching themes to writing effective code:
-
- 1. **Be consistent.** With things like formatting, and casing, arguments as to
-    which is better tend to end up being mostly subjective and impossible to
+ 1. **Be consistent.** When it comes to things like formatting, and casing,
+    arguments about to which is better end up subjective and impossible to
```
Comment (kwalrath) [inline src/_guides/language/effective-dart/index.md]: to which -> which
end up  -> are

File: src/_guides/language/effective-dart/index.md [hunk 3]
```diff
@@ -14,24 +14,21 @@ nextpage:
...
-    the same statements and expressions as C, Java, JavaScript and other
-    languages. But we created Dart because there is a lot of room to improve on
-    what those languages offer. We added a bunch of features, from string
-    interpolation to initializing formals to help you express your intent more
-    simply and easily.
+ 2. **Be brief.** Dart was designed to be familiar, so it inherits many of the
+    same statements and expressions as C, Java, JavaScript and other languages.
+    But we created Dart because there is a lot of room to improve on what those
+    languages offer. We added a bunch of features, from string interpolation to
+    initializing formals to help you express your intent more simply and easily.
```
Comment (kwalrath) [inline src/_guides/language/effective-dart/index.md]: formals -> formals,

File: src/_guides/language/effective-dart/index.md [hunk 4]
```diff
@@ -86,13 +83,11 @@ Each guide is broken into a few sections. Sections contain a list of guidelines.
...
-will almost never be a valid reason to stray from them.
+  will almost never be a valid reason to stray from them.
 
 * **DON'T** guidelines are the converse: things that are almost never a good
-idea. Fortunately, we don't have many of these in Dart. Rules like these in
-other languages help avoid problems caused by historical baggage. Dart is
-new enough that we can just fix those problems directly instead of putting
-up ropes around them.
+  idea. Hopefully, we don't have as many of these as other languages do because
+  we have less historical baggage to route around.
```
Comment (kwalrath) [inline src/_guides/language/effective-dart/index.md]: delete " to route around"

File: src/_guides/language/effective-dart/index.md [hunk 5]
```diff
@@ -115,19 +110,23 @@ readable and maintainable code.
 To keep the guidelines brief, we use a few shorthand terms to refer to different
 Dart constructs.
 
-* A **member** is any definition or declaration in a library. This includes
-  top-level variables, getters, setters, and functions. It also includes
-  instance and static members inside classes: fields, getters, setters, methods,
-  and operators.
+* A **library member** is a top-level field, getter, setter or function.
```
Comment (kwalrath) [inline src/_guides/language/effective-dart/index.md]: setter or -> setter, or

(Google is a serial-comma shop, but I was going to let non-serial commas slide until I saw that the next paragraph uses one in a very similar place!)

File: src/_guides/language/effective-dart/index.md [hunk 6]
```diff
@@ -115,19 +110,23 @@ readable and maintainable code.
...
+  Basically, anything at the top level that isn't a type.
+
+* A **class member** is a constructor, field, getter, setter, function, or
+  operator declared inside a class. They can be instance or static, abstract or
```
Comment (kwalrath) [inline src/_guides/language/effective-dart/index.md]: They -> Class members

File: src/_guides/language/effective-dart/index.md [hunk 7]
```diff
@@ -115,19 +110,23 @@ readable and maintainable code.
...
+  concrete.
+
+* A **member** is either a library member or class member.
```
Comment (kwalrath) [inline src/_guides/language/effective-dart/index.md]: or -> or a

File: src/_guides/language/effective-dart/index.md [hunk 8]
```diff
@@ -115,19 +110,23 @@ readable and maintainable code.
...
 
-* A **variable** is any variable declaration, local or top-level. This also
-  includes member parameters.
+* A **variable** when used generally, refers to top-level variables,
```
Comment (kwalrath) [inline src/_guides/language/effective-dart/index.md]: when -> , when

File: src/_guides/language/effective-dart/index.md (hunk 1 above)
Comment (munificent) [reply src/_guides/language/effective-dart/index.md]: Done.

File: src/_guides/language/effective-dart/index.md (hunk 2 above)
Comment (munificent) [reply src/_guides/language/effective-dart/index.md]: Done.

File: src/_guides/language/effective-dart/index.md (hunk 3 above)
Comment (munificent) [reply src/_guides/language/effective-dart/index.md]: Done.

File: src/_guides/language/effective-dart/index.md (hunk 4 above)
Comment (munificent) [reply src/_guides/language/effective-dart/index.md]: Done.

File: src/_guides/language/effective-dart/index.md (hunk 5 above)
Comment (munificent) [reply src/_guides/language/effective-dart/index.md]: Do,ne.

File: src/_guides/language/effective-dart/index.md (hunk 6 above)
Comment (munificent) [reply src/_guides/language/effective-dart/index.md]: Done.

File: src/_guides/language/effective-dart/index.md (hunk 7 above)
Comment (munificent) [reply src/_guides/language/effective-dart/index.md]: Done.

File: src/_guides/language/effective-dart/index.md (hunk 8 above)
Comment (munificent) [reply src/_guides/language/effective-dart/index.md]: , done.

**Pull request merged**